src/app/Common/DAL/UpdateDocumentOperators/IncrementOperator.cs

```diff
@@ -7,11 +9,29 @@ namespace Common.DAL.UpdateDocumentOperators
 {
     public class IncrementOperator<TDocument, TItem>: IUpdateOperator<TDocument> where TDocument: BaseDocument
     {
+        private static readonly HashSet<Type> _allowedTypes;
         private readonly Expression<Func<TDocument, TItem>> _field;
         private readonly TItem _incrementByValue;
 
+        static IncrementOperator()
+        {
+            var types = new[]
+            {
+                typeof(byte), typeof(sbyte), typeof(short), typeof(ushort), typeof(int), typeof(uint), typeof(long),
+                typeof(ulong), typeof(float), typeof(double), typeof(decimal),
+            };
+            
+            _allowedTypes =
+                new HashSet<Type>(types.Concat(types.Select(t => typeof(Nullable<>).MakeGenericType(t))));
+        }
+
         public IncrementOperator(Expression<Func<TDocument, TItem>> field, TItem incrementByValue)
         {
+            if (!_allowedTypes.Contains(field.ReturnType) || !_allowedTypes.Contains(incrementByValue.GetType()))

```

Comment:
I think these are two identical checks. Because field.ReturnType == typeof(TItem) and incrementByValue.GetType() == typeof(TItem). You can leave one check. For example:
`!_allowedTypes.Contains(typeof(TItem))`